### PR TITLE
[script.tag-generator] 0.7.0

### PR DESCRIPTION
**Pull request by wellspokenman**
### Description
Leia support and bugfixes
### Checklist:

- [X] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [X] Each add-on submission should be a single commit with using the following style: [script.foo.bar] 1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practise but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-scripts/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

**Comment by wellspokenman** [comment]
The first few logs were pretty clear - not sure why travis doesn't like this one.

File: script.tag-generator/addon.xml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<addon id="script.tag-generator"
+       name="Tag Generator"
+       version="0.7.0"
+       provider-name="wellspokenman">
+  <requires>
+    <import addon="xbmc.python" version="2.26.0"/>
+    <import addon="script.module.simplejson" version="3.3.0"/>
```
Comment (enen92) [inline script.tag-generator/addon.xml]: Not needed anymore, please use only `json`

File: script.tag-generator/addon.xml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    <import addon="script.module.requests_oauthlib" version="0.7.0"/>
+    <import addon="script.module.beautifulsoup4" version="4.5.3"/>
+  </requires>
+  <extension point="xbmc.service" library="main.py" />
+  <extension point="xbmc.addon.metadata">
+    <platform>all</platform>
+    <summary lang="en_GB">IMDB, Trakt, Oscar and Stand-up Tag Generator</summary>
+    <description lang="en_GB">Helps you create Smart Playlists from IMDb/Trakt Watchlists by tagging local movies found in specified lists. Also creates a stand-up comedy tag based on Wikipedia comedians, as there is no genre for stand-up. Specify your IMDb/Trakt watchlists then choose your desired Kodi tag for each list. Runs as a service in the background at a specified interval, and also allows for manual creation. For Trakt, you can add lists from your own account, from public accounts, or shared lists from anyone you are friends with (people you follow, who follow you back). Oscar tagging lets you tag winners and/or nominees with each award for each movie in the following format 'Winner: Best Motion Picture'. </description>
+	<license>GNU GENERAL PUBLIC LICENSE. Version 2, June 1991</license>
```
Comment (enen92) [inline script.tag-generator/addon.xml]: `GPL-2.0-only`  instead of the full text if possible

File: script.tag-generator/changelog.txt [hunk 3]
```diff
@@ -0,0 +1,44 @@
+v0.7 (December 12th, 2018)
```
Comment (enen92) [inline script.tag-generator/changelog.txt]: Please drop from the submission if possible

File: script.tag-generator/main.py [hunk 4]
```diff
@@ -0,0 +1,643 @@
+from bs4 import BeautifulSoup
+import datetime
+import json
+import os
+import re
+import requests
+import simplejson
```
Comment (enen92) [inline script.tag-generator/main.py]: drop

File: script.tag-generator/main.py [hunk 5]
```diff
@@ -0,0 +1,643 @@
...
+import sys
+import xbmc
+import xbmcgui
+import xbmcaddon
+
+sys.path.append(os.path.join(os.path.abspath(os.path.dirname(__file__)), "resources", "lib"))
```
Comment (enen92) [inline script.tag-generator/main.py]: This is really a bad practice. Why is it needed? Can't you just import your modules from.resources.lib?

File: script.tag-generator/main.py [hunk 6]
```diff
@@ -0,0 +1,643 @@
...
+def ifcancel():
+    if len(sys.argv) == 2:
+        if (pDialog.iscanceled()):
+            debuglog("TAG-GEN: Cancel received from Kodi dialog, exiting.")
+            sys.exit(0)
+
+
+def debuglog(string):
+    if c_debug:
+        xbmc.log(msg=string, level=xbmc.LOGNOTICE)
```
Comment (enen92) [inline script.tag-generator/main.py]: Default level must be xbmc.LOGDEBUG

File: script.tag-generator/main.py [hunk 7]
```diff
@@ -0,0 +1,643 @@
...
+    if wipeout:
+        pDialog.update(0, _getstr(30004), " ", " ")
+    elif len(sys.argv) == 2:
+        pDialog.update(0, _getstr(30005), " ", " ")
+    # json strings -> unicode -> json
+    json_query = xbmc.executeJSONRPC('{"jsonrpc": "2.0", "method": "VideoLibrary.GetMovies", "params": {"properties" : '
+                                     '["title","studio","tag","imdbnumber","year"], "sort": {"order": "ascending", '
+                                     '"method": "label", "ignorearticle": true}}, "id": "libMovies"}')
+    json_query = unicode(json_query, 'utf-8', errors='ignore')
+    jsonobject = simplejson.loads(json_query)
```
Comment (enen92) [inline script.tag-generator/main.py]: json.loads

File: script.tag-generator/main.py [hunk 8]
```diff
@@ -0,0 +1,643 @@
...
+    elif wipeout:
+        dialog = xbmcgui.Dialog()
+        ok = dialog.ok("Tag Generator", _getstr(30034) + str(wipealltags()) + _getstr(30035))
+        debuglog("TAG-GEN: Wipeout arg received, exiting after single execution.")
+        sys.exit(0)
+    else:
+        xbmc.log(msg="TAG-GEN: Sleeping for " + str(c_refresh) + " hours", level=xbmc.LOGNOTICE)
+        while not monitor.abortRequested():
+            if (datetime.datetime.now() - start_time).total_seconds() < service_interval:
+                xbmc.sleep(micronap)
```
Comment (enen92) [inline script.tag-generator/main.py]: use xbmc.Monitor.waitForAbort() while sleeping in long loops

File: script.tag-generator/resources/lib/trakt/errors.py [hunk 9]
```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
```
Comment (enen92) [inline script.tag-generator/resources/lib/trakt/errors.py]: Isn't this already available from script.module.trakt? @razzeee

Comment (Razzeee) [reply script.tag-generator/resources/lib/trakt/errors.py]: I think this is Pytrakt, I'm using Trakt.py

File: script.tag-generator/main.py (hunk 4 above)
Comment (wellspokenman) [reply script.tag-generator/main.py]: > 
> 
> drop
replaced simplejson calls with json and dropped the import of simplejson

File: script.tag-generator/main.py (hunk 5 above)
Comment (wellspokenman) [reply script.tag-generator/main.py]: I don't like it either, but if I don't import it this way the trakt module doesn't work. I've tried:
```
import resources.lib.trakt as trakt
from trakt import users as traktusers
```
and
```
import resources.lib.trakt as trakt
from resources.lib.trakt import users as traktusers
```
The latter results in an error in the trakt users module, which isn't written by me. I prefer hacky to editing someone else's code.

File: script.tag-generator/main.py (hunk 8 above)
Comment (wellspokenman) [reply script.tag-generator/main.py]: tested and implemented - been meaning to do that for a while; thanks.

File: script.tag-generator/main.py (hunk 7 above)
Comment (wellspokenman) [reply script.tag-generator/main.py]: done

File: script.tag-generator/main.py (hunk 6 above)
Comment (wellspokenman) [reply script.tag-generator/main.py]: oops - done

File: script.tag-generator/changelog.txt (hunk 3 above)
Comment (wellspokenman) [reply script.tag-generator/changelog.txt]: done - i'll keep it in my repo; no need to submit to the official

File: script.tag-generator/addon.xml (hunk 2 above)
Comment (wellspokenman) [reply script.tag-generator/addon.xml]: done

File: script.tag-generator/addon.xml (hunk 1 above)
Comment (wellspokenman) [reply script.tag-generator/addon.xml]: done

**Comment by wellspokenman** [comment]
> 
> 
> See comments. If the trakt module is already available in Kodi as a script module it should be imported and used as a dependency

As far as I can tell there isn't an in-built kodi dependency I can call for this. All other requests should be satisfied in the new commit: https://github.com/xbmc/repo-scripts/pull/1252/commits/7df21268be811e08b248d21132eb898b5fda0845

File: script.tag-generator/main.py [hunk 10]
```diff
@@ -637,7 +636,9 @@ def write_award_tags(medialist, tag_winners, tag_nominees):
         xbmc.log(msg="TAG-GEN: Sleeping for " + str(c_refresh) + " hours", level=xbmc.LOGNOTICE)
         while not monitor.abortRequested():
             if (datetime.datetime.now() - start_time).total_seconds() < service_interval:
-                xbmc.sleep(micronap)
+                if monitor.waitForAbort(micronap):
```
Comment (enen92) [inline script.tag-generator/main.py]: This accepts seconds, is it correct?

Comment (wellspokenman) [reply script.tag-generator/main.py]: oops - thanks for catching that.

**Comment by wellspokenman** [comment]
@enen92 I don't see this version from my Kodi leia interface. Usually it shows up after the PR is approved - is there something else I should do?

**Comment by enen92** [comment]
@wellspokenman followed this to others since I can't see any reason for the failure.

**Comment by enen92** [comment]
@wellspokenman sorry for the huge delay. Hopefully, @kib gave a hand with this. Cause is here:
https://github.com/xbmc/repo-scripts/pull/1299

I took the chance to push an update to your addon, please fix it upstream too.

**Pull request merged**